Stratis.Bitcoin/Features/Wallet/WalletManager.cs

```diff
@@ -292,221 +292,234 @@ public HdAddress GetUnusedAddress(WalletAccountReference accountReference)
+        }
+
+        /// <summary>
+        /// Gets the hash of the oldest block received by the wallets.
+        /// </summary>
+        /// <returns></returns>
+        public uint256 LastReceivedBlockHash()
+        {
+            if (!this.Wallets.Any())
+            {
+                return this.chain.Tip.HashBlock;
+            }
 
             var lastBlockSyncedHash = this.Wallets.Select(w => w.AccountsRoot.SingleOrDefault(a => a.CoinType == this.coinType))
                 .Where(w => w != null)
                 .OrderBy(o => o.LastBlockSyncedHeight)
                 .FirstOrDefault()?.LastBlockSyncedHash;
-			Guard.Assert(lastBlockSyncedHash != null);
-			return lastBlockSyncedHash;
-		}
-
-		/// <inheritdoc />
-		public List<UnspentInfo> GetSpendableTransactions(int confirmations = 0)
-		{
-			var outs = new List<UnspentInfo>();
-			var accounts = this.Wallets.SelectMany(wallet => wallet.AccountsRoot.Single(a => a.CoinType == this.coinType).Accounts);
-
-			var currentHeight = this.chain.Tip.Height;
-
-			// this will take all the spendable coins 
-			// and keep the reference to the HDAddress
-			// so later the private key can be calculated 
-			// for the given unspent outputs 
-
-			foreach (var account in accounts)
-			{
-				foreach (var externalAddress in account.ExternalAddresses)
-				{
-					var unspent = externalAddress.UnspentTransactions().Where(a => currentHeight - (a.BlockHeight ?? currentHeight) >= confirmations).ToList();
-					if (unspent.Any())
-					{
-						outs.Add(new UnspentInfo
-						{
-							Account = account,
-							Address = externalAddress,
-							Transactions = unspent
-						});
-					}
-
-				}
-				foreach (var internalAddress in account.InternalAddresses)
-				{
-					var unspent = internalAddress.UnspentTransactions().Where(a => currentHeight - (a.BlockHeight ?? currentHeight) >= confirmations).ToList();
-					if (unspent.Any())
-					{
-						outs.Add(new UnspentInfo
-						{
-							Account = account,
-							Address = internalAddress,
-							Transactions = unspent
-						});
-					}
-				}
-			}
-
-			return outs;
-		}
-
-		/// <inheritdoc />
-		public ISecret GetKeyForAddress(string password, HdAddress address)
-		{
-			// TODO: can we have more then one wallet per coins?
-			var walletTree = this.Wallets.First();
-			// get extended private key
-			var privateKey = Key.Parse(walletTree.EncryptedSeed, password, walletTree.Network);
-			var seedExtKey = new ExtKey(privateKey, walletTree.ChainCode);
-			ExtKey addressExtKey = seedExtKey.Derive(new KeyPath(address.HdPath));
-			BitcoinExtKey addressPrivateKey = addressExtKey.GetWif(walletTree.Network);
-			return addressPrivateKey;
-		}
+            Guard.Assert(lastBlockSyncedHash != null);
+            return lastBlockSyncedHash;
+        }
+
+        /// <inheritdoc />
+        public List<UnspentInfo> GetSpendableTransactions(int confirmations = 0)
```

Comment:
I have already talked about trying it with dan however I wanted to do that in a different PR.